src/controllers/sign.out.controller.ts

```diff
@@ -29,17 +28,34 @@ export const post = (req: Request, res: Response, next: NextFunction) => {
     logger.debugRequest(req, `POST ${config.SIGN_OUT_PAGE}`);
     const previousPage = req.body["previousPage"];
 
-    if (!previousPage.includes(config.REGISTER_AN_OVERSEAS_ENTITY_URL)){
+    if (!previousPage.startsWith(config.REGISTER_AN_OVERSEAS_ENTITY_URL)){
       throw createAndLogErrorRequest(req, `${previousPage} page is not part of the journey!`);
     }
 
     if (req.body["sign_out"] === 'yes') {
       return res.redirect(config.ACCOUNTS_SIGN_OUT_URL);
     }
 
-    return res.redirect(previousPage);
+    return safeRedirect(res, previousPage);
   } catch (error) {
     logger.errorRequest(req, error);
     next(error);
   }
 };
+
+function getPreviousPageUrl(req: Request) {
+  const headers = req.rawHeaders;
+  const absolutePreviousPageUrl = headers.filter(item => item.includes(config.REGISTER_AN_OVERSEAS_ENTITY_URL))[0];
+
+  // Don't attempt to determine a relative previous page URL if no absolute URL is found
+  if (!absolutePreviousPageUrl) {
+    return absolutePreviousPageUrl;
+  }
+
+  const startingIndexOfRelativePath = absolutePreviousPageUrl.indexOf(config.REGISTER_AN_OVERSEAS_ENTITY_URL);
+  const relativePreviousPageUrl = absolutePreviousPageUrl.substring(startingIndexOfRelativePath);
+
+  logger.debug(`Relative previous page URL is ${relativePreviousPageUrl}`);
```

Comment:
can you use debugRequest here and use the req param

Reply:
Done.